newIDE/app/src/Utils/GDevelopServices/Project.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+import axios from 'axios';
+import {
+  GDevelopProjectApi,
+  GDevelopProjectResourcesStorage,
+} from './ApiConfigs';
+import { type AuthenticatedUser } from '../../Profile/AuthenticatedUserContext';
+
+export const CLOUD_PROJECT_NAME_MAX_LENGTH = 50;
+
+const projectResourcesClient = axios.create({
+  withCredentials: true,
+});
+
+type CloudProject = {|
+  id: string,
+  name: string,
+  createdAt: string,
+  currentVersion?: string,
+  deletedAt?: string,
+|};
+
+export type CloudProjectWithUserAccessInfo = {|
+  ...CloudProject,
+  lastModifiedAt: string,
+|};
+
+const refetchCredentialsForProjectAndRetryIfFailed = async <T>(
```

Comment:
maybe replace `Failed` with `Unauthorized` as this is what really happens?

Reply:
Good idea, I did the change